modules/gapi/test/cpu/gapi_ocv_stateful_kernel_tests.cpp

```diff
@@ -167,7 +224,114 @@ TEST(StatefulKernel, InvalidReallocatingKernel)
     cv::GComputation comp(cv::GIn(in), cv::GOut(out));
 
     EXPECT_THROW(comp.apply(in_mat, out_mat, cv::compile_args(pkg)), std::logic_error);
+}
+
+namespace
+{
+    void compareBackSubResults(const cv::Mat &actual, const cv::Mat &expected,
+                                       const int diffPercent)
+    {
+        GAPI_Assert(actual.size() == expected.size());
+        int allowedNumDiffPixels = actual.size().area() * diffPercent / 100;
+
+        cv::Mat diff;
+        cv::absdiff(actual, expected, diff);
+        #undef countNonZero
+        int numDiffPixels = cv::countNonZero(diff);
+
+        EXPECT_GT(allowedNumDiffPixels, numDiffPixels);
+    }
+} // anonymous namespace
+
+TEST(StatefulKernel, StateIsInitViaCompArgs)
+{
+    cv::Mat frame(1080, 1920, CV_8UC3),
+            gapiForeground,
+            ocvForeground;
+
+    cv::randu(frame, cv::Scalar(0, 0, 0), cv::Scalar(255, 255, 255));
```

Comment:
Y not play a file instead? `opencv_extra` has some.

Reply:
This, again, can be done separately if we decide to have Background Subtractor at the API level.

...And there is already a demand in it. It may be a next task to do

Reply:
Ok I see it below. Great!